lib/simplecov/result_merger.rb

```diff
@@ -87,6 +94,25 @@ def store_result(result)
         end
         true
       end
+
+    private
+
+      # Ensure only one process is reading or writing the resultset at any
+      # given time
+      def synchronize_resultset
+        # make it reentrant
+        return yield if defined?(@resultset_locked) && @resultset_locked
```

Comment:
It does, but it issues a ruby warning `instance variable @resultset_locked not initialized`, which fails the build, thanks to spec/support/fail_rspec_on_ruby_warning.rb